src/code/TwoHeadArena.c

```diff
@@ -1,140 +1,135 @@
+/**
+ * @file TwoHeadArena.c
+ *
+ * This file implements a simple double-ended stack allocator for general purpose.
+ *
+ * A double-ended stack allocator accepts allocations at either the "head" or "tail" of it's allotted memory region.
+ * While in general this type of allocator could accept deallocations on the most recently allocated block at either
+ * end, this implementation does not support any individual deallocations; the only provided way to deallocate anything
+ * is to reset the entire arena, deallocating everything. This scheme is most applicable to allocating similar data
+ * with identical lifetime.
+ */
 #include "global.h"
 
-void THGA_Ct(TwoHeadGfxArena* thga, Gfx* start, u32 size) {
-    THA_Ct((TwoHeadArena*)thga, start, size);
-}
-
-void THGA_Dt(TwoHeadGfxArena* thga) {
-    THA_Dt((TwoHeadArena*)thga);
-}
-
-u32 THGA_IsCrash(TwoHeadGfxArena* thga) {
-    return THA_IsCrash((TwoHeadArena*)thga);
-}
-
-void THGA_Init(TwoHeadGfxArena* thga) {
-    THA_Init((TwoHeadArena*)thga);
-}
-
-s32 THGA_GetSize(TwoHeadGfxArena* thga) {
-    return THA_GetSize((TwoHeadArena*)thga);
-}
-
-Gfx* THGA_GetHead(TwoHeadGfxArena* thga) {
-    return THA_GetHead((TwoHeadArena*)thga);
-}
-
-void THGA_SetHead(TwoHeadGfxArena* thga, Gfx* start) {
-    THA_SetHead((TwoHeadArena*)thga, start);
-}
-
-Gfx* THGA_GetTail(TwoHeadGfxArena* thga) {
-    return THA_GetTail((TwoHeadArena*)thga);
-}
-
-Gfx* THGA_AllocStartArray8(TwoHeadGfxArena* thga, u32 count) {
-    return THA_AllocStart((TwoHeadArena*)thga, count * 8);
-}
-
-Gfx* THGA_AllocStart8(TwoHeadGfxArena* thga) {
-    return THGA_AllocStartArray8(thga, 1);
-}
-
-Gfx* THGA_AllocStart8Wrapper(TwoHeadGfxArena* thga) {
-    return THGA_AllocStart8(thga);
-}
-
-Gfx* THGA_AllocEnd(TwoHeadGfxArena* thga, u32 size) {
-    return THA_AllocEnd((TwoHeadArena*)thga, size);
-}
-
-Gfx* THGA_AllocEndArray64(TwoHeadGfxArena* thga, u32 count) {
-    return THGA_AllocEnd(thga, count * 0x40);
-}
-
-Gfx* THGA_AllocEnd64(TwoHeadGfxArena* thga) {
-    return THGA_AllocEnd(thga, 0x40);
-}
-
-Gfx* THGA_AllocEndArray16(TwoHeadGfxArena* thga, u32 count) {
-    return THGA_AllocEnd(thga, count * 0x10);
-}
-
-Gfx* THGA_AllocEnd16(TwoHeadGfxArena* thga) {
-    return THGA_AllocEnd(thga, 0x10);
-}
-
 void* THA_GetHead(TwoHeadArena* tha) {
     return tha->head;
 }
 
-void THA_SetHead(TwoHeadArena* tha, void* start) {
-    tha->head = start;
+void THA_SetHead(TwoHeadArena* tha, void* newHead) {
+    tha->head = newHead;
 }
 
 void* THA_GetTail(TwoHeadArena* tha) {
     return tha->tail;
 }
 
-void* THA_AllocStart(TwoHeadArena* tha, u32 size) {
+/**
+ * Allocates to the head of the Two Head Arena. The allocation will not have any alignment guarantees.
+ */
+void* THA_AllocStart(TwoHeadArena* tha, size_t size) {
```

Comment:
i havent done a thorough review at all, just during my initial skim:
I read this as "start the allocation" even tho i know the intention is start and end/front and back
we should prob use head and tail? since we're already using those terms in the functions above

Reply:
I also realize it was already named this, just never noticed before i guess

Reply:
I wasn't too sure what sort of direction to take names in so if you spot anything else that wasn't renamed but probably should be that'd be great